File: src/contests/cg/puzzle/swift/max-area.swift
// 🎮 CodinGame Puzzle - max-area
// https://www.codingame.com/training/easy/max-area

_ = readLine()
let a = readLine()!.split(separator: " ").map { Int($0)! }
var left = 0
var right = a.count - 1
var best = 0
while left < right {
    let h = min(a[left], a[right])
    let area = h * (right - left)
    if area > best {
        best = area
    }
    if a[left] < a[right] {
        left += 1
    } else {
        right -= 1
    }
}
print(best)

File: src/contests/cg/puzzle/clojure/max-area.clj
;; 🎮 CodinGame Puzzle - max-area
;; https://www.codingame.com/training/easy/max-area

(ns Solution (:gen-class))

(defn -main [& _]
  (read-line)
  (let [a (mapv #(Long/parseLong %) (.split (.trim (read-line)) "\\s+"))]
    (loop [left 0 right (dec (count a)) best 0]
      (if (< left right)
        (let [h (min (nth a left) (nth a right))
              nb (max best (* h (- right left)))]
          (if (< (nth a left) (nth a right))
            (recur (inc left) right nb)
            (recur left (dec right) nb)))
        (println best))))
  (flush))
